documentation/src/docs/asciidoc/release-notes/release-notes-6.0.0-M1.adoc

```diff
@@ -63,7 +63,12 @@ repository on GitHub.
 [[release-notes-6.0.0-M1-junit-jupiter-deprecations-and-breaking-changes]]
 ==== Deprecations and Breaking Changes
 
-* ❓
+* The `JRE` enum constants for `JAVA_8` to `JAVA_16` have been deprecated because they can
+  no longer be used at runtime since `JAVA_17` is the new baseline.
+* `@EnabledForJreRange` and `@DisabledForJreRange` now use `JAVA_17` as their default
+  `min` value.
+* `@EnabledOnJre`, `@DisabledOnJre`, `@EnabledForJreRange`, and `@DisabledForJreRange` now
+  fail if a JRE version less than 17 is specified.
```

Comment:
I don't think we can do that, since that would break existing code.

In other words, deprecating the unsupported enum constants should suffice: people will see the warnings and update their code accordingly. But... existing code should not break.

For example, `@EnabledForJreRange(min = JAVA_9, max = JAVA_21)` should still logically work with a Java 17 baseline, and the deprecation of `JAVA_9` will encourage the user to migrate that declaration to `@EnabledForJreRange(min = JAVA_17, max = JAVA_21)`.

Reply:
We could make the strictness of the validation configurable via a configuration parameter... :thinking:

Reply:
> We could relax the validation to only fail if a range contains no supported runtime version (so 9-21 would be allowed because contains 17-21).

Yes, let's do that.

If we can determine that the range is now obsolete, throw an exception.

If we can determine that the range is still valid (even if too broad), do not throw an exception.

Reply:
> We could make the strictness of the validation configurable via a configuration parameter... 🤔

Mmmm... yeah, we _could_ do that, but I'd rather avoid too many behavioral configuration flags if possible.

Maybe we should just see how the community reacts to "rejection of no longer applicable ranges".